engineio/src/transports/websocket.rs

```diff
@@ -1,122 +1,69 @@
-use crate::error::{Error, Result};
-use crate::packet::Packet;
-use crate::packet::PacketId;
-use crate::transport::Transport;
-use bytes::{BufMut, Bytes, BytesMut};
-use std::borrow::Cow;
-use std::str::from_utf8;
-use std::sync::{Arc, Mutex, RwLock};
-use websocket::{
-    client::Url, dataframe::Opcode, header::Headers, receiver::Reader, sync::stream::TcpStream,
-    sync::Writer, ws::dataframe::DataFrame, ClientBuilder as WsClientBuilder, Message,
+use crate::{
+    async_transports::{transport::AsyncTransport, websocket::AsyncWebsocketTransport},
+    error::Result,
+    header::HeaderMap,
+    transport::Transport,
 };
+use bytes::Bytes;
+use std::sync::Arc;
+use tokio::runtime::Runtime;
+use url::Url;
 
 #[derive(Clone)]
 pub struct WebsocketTransport {
-    sender: Arc<Mutex<Writer<TcpStream>>>,
-    receiver: Arc<Mutex<Reader<TcpStream>>>,
-    base_url: Arc<RwLock<url::Url>>,
+    runtime: Arc<Runtime>,
+    inner: Arc<AsyncWebsocketTransport>,
 }
 
 impl WebsocketTransport {
     /// Creates an instance of `WebsocketTransport`.
-    pub fn new(base_url: Url, headers: Option<Headers>) -> Result<Self> {
+    pub fn new(base_url: Url, headers: Option<HeaderMap>) -> Result<Self> {
```

Comment:
suggestion: This is a change in the public interface, will require a semver minor bump.

Reply:
Yeah... this is difficult, but the `Headers` type comes from the hyper version that is suspect to a security issue. So using our own type might just be safer... Or we use the [`http::header::HeaderMap`](https://docs.rs/http/latest/http/header/struct.HeaderMap.html) type.

Reply:
I decided to use the more popular `http::HeaderMap` and added the proper conversions for it.